alf/networks/projection_networks.py

```diff
@@ -82,16 +84,24 @@ def __init__(self,
 
         self._output_shape = output_shape
         self._projection_layer = projection_layer
+        self._disable_amp = disable_amp
+        self._amp_dtype = alf.get_config_value('TrainerConfig.amp_dtype')
 
     def forward(self, inputs, state=()):
-        logits, state = self._projection_layer(inputs, state)
-        logits = logits.reshape(inputs.shape[0], *self._output_shape)
-        if len(self._output_shape) > 1:
-            return td.Independent(
-                td.Categorical(logits=logits),
-                reinterpreted_batch_ndims=len(self._output_shape) - 1), state
-        else:
-            return td.Categorical(logits=logits), state
+        amp_enabled = torch.is_autocast_enabled()
+        if self._disable_amp and amp_enabled:
+            inputs = alf.layers.to_float32(inputs)
+            amp_enabled = False
```

Comment:
I wonder did you have negative experience with bfloat16 for this network?

Reply:
yes